File: src/main/java/hacker/solution/IsPrime.java
package hacker.solution;

import java.math.BigInteger;
import java.util.Scanner;

public class IsPrime {
    private static final Scanner scanner = new Scanner(System.in);
    public static void main(String[] args) {
        String s = scanner.nextLine();
        try {
            BigInteger num = new BigInteger(s);
            boolean flag = false;
            BigInteger two = new BigInteger("2");
            if (!num.divide(two).equals(BigInteger.ZERO) && num.isProbablePrime(80)) {
//                for (BigInteger i = new BigInteger("3"); i.multiply(i).compareTo(num) < 1; i = i.add(two))
//                {
//                    if (BigInteger.ZERO.equals(num.mod(i))) {
//                        flag = false;
//                        break;
//                    }
//                }
                flag = true;
            }
            if (flag && num != BigInteger.ONE)
                System.out.println("prime");
            else
                System.out.println("not prime");
        } catch (NumberFormatException nfe) {
            System.out.println("Problem parsing input="+s+": "+nfe.getMessage());
        }
        scanner.close();
    }
}

File: src/main/kotlin/hacker/solutionk/IsPrime.kt
package hacker.solutionk

import java.math.BigInteger
import java.util.*

object IsPrime {
    private val scanner = Scanner(System.`in`)
    @JvmStatic
    fun main(args: Array<String>) {
        val s = scanner.nextLine()
        try {
            val num = BigInteger(s)
            var flag = false
            val two = BigInteger("2")
            if (num.divide(two) != BigInteger.ZERO && num.isProbablePrime(80)) {
                flag = true
            }
            if (flag && num !== BigInteger.ONE)
                println("prime")
            else
                println("not prime")
        } catch (nfe: NumberFormatException) {
            println("Problem parsing input=" + s + ": " + nfe.message)
        }
        scanner.close()
    }
}
